src/hooks/__internal__/useFloating/useFloating.ts

```diff
@@ -8,14 +8,12 @@ import {
   Placement,
 } from "@floating-ui/dom";
 
-type CustomRefObject<T> = {
-  current?: T | null;
-};
-
 export interface UseFloatingProps {
   isOpen?: boolean;
-  reference: CustomRefObject<HTMLElement | null>;
-  floating: React.RefObject<HTMLElement | null>;
```

Comment:
These types caused problems while using `SelectList` in tests. Also refactored this elsewhere in the codebase for consistency.